File: server/src/main/java/com/example/schoolsite/services/UserDetailsServiceImpl.java
package com.example.schoolsite.services;

import com.example.schoolsite.entity.User;
import com.example.schoolsite.workWithDatabase.repo.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;

public class UserDetailsServiceImpl implements UserDetailsService {

    @Autowired
    UserRepository userRepository;

    @Override
    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
        User user = userRepository
                .findByLogin(username)
                .orElseThrow(()->new UsernameNotFoundException("User not found with  username: " + username));
        return UserDetailsImpl.build(user);
    }
}

File: server/src/main/java/com/example/schoolsite/services/impl/UserDetailsServiceImpl.kt
package com.example.schoolsite.services.impl

import com.example.schoolsite.entity.User

@Service
@RequiredArgsConstructor
class UserDetailsServiceImpl : UserDetailsService {
    private val userRepository: UserRepository? = null
    @Override
    @Throws(UsernameNotFoundException::class)
    fun loadUserByUsername(username: String): UserDetails {
        val user: User = userRepository
                .findByLogin(username)
                .orElseThrow { UsernameNotFoundException("User not found with  username: $username") }
        return UserDetailsImpl.build(user)
    }
}